core/control-plane/control-plane-aggregate-services/src/main/java/org/eclipse/edc/connector/service/contractnegotiation/ContractNegotiationProtocolServiceImpl.java

```diff
@@ -194,6 +204,17 @@ private ServiceResult<ContractNegotiation> validateRequest(ClaimToken claimToken
             return ServiceResult.success(negotiation);
         }
     }
+    
+    @NotNull
+    private ServiceResult<ContractNegotiation> validateGetRequest(ClaimToken claimToken, String id, ContractNegotiation negotiation) {
+        var result = validationService.validateRequest(claimToken, negotiation);
+        if (result.failed()) {
+            // should return not found if counter-party not authorized
+            return ServiceResult.notFound(format("No negotiation with id %s found", id));
```

Comment:
if this failure is the same as the one 153, to avoid the duplication this method could just return the `ContractNegotiation` (so `null` in this branch and `negotiation` in the `else`), then in the mail optional flow a `.map(StatusResult::success)` can be added,

Reply:
good point, refactored it